doc/storage/api/api.rst

```diff
@@ -379,6 +402,9 @@ These definitions must be defined in the header file :file:`psa/protected_storag
         * The ``uid`` is ``0``.
 
         * The operation failed because caller cannot access some or all of the memory in the range [``p_data``, ``p_data + data_length - 1``].
+        
+        * the uid exists and ``data_length`` is greater then ```capacity`` 
```

Comment:
I think the intention was always that if a file was created with capacidy = x then any set with size < x will succeed. 

I thought the initial intention was that if size > capacity - for either set or set_extended - then it fials. 

If the intention was that it might succeed, but cannot be guaranteed, that is also fine.

Reply:
> I think the intention was always that if a file was created with capacity = x then any set with size < x will succeed.

I don't know what the intent was, but I find this requirement surprising. In a log-structured filesystem, overwriting a file with less content takes some disk space. Even deleting a file can require some disk space in some filesystems.